### PR TITLE
msal-react alpha.0

### PR DESCRIPTION
**Pull request by tnorling**
Release branch for msal-react alpha.0

### CODE REVIEW

File: lib/msal-react/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+`msal-react` is under development. **We do not recommend using this in a production environment yet**.
+
+| [AAD Docs](https://aka.ms/aaddevv2) | [Support](README.md#community-help-and-support) | [Samples](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/samples)
+| --- | --- | --- |
+
+1. [About](#about)
+1. [FAQ](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/FAQ.md)
+1. [Prerequisites](#prerequisites)
+1. [Installation](#installation)
+1. [Getting Started](#https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/getting-started.md)
```
Comment (jo-arroyo) [inline lib/msal-react/README.md:12]:
```suggestion
1. [Getting Started](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/getting-started.md)
```

File: lib/msal-react/docs/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
+# Getting Started
+
+1. [Initialization](#Intialization)
```
Comment (jo-arroyo) [inline lib/msal-react/docs/getting-started.md:3]:
```suggestion
1. [Initialization](#Initialization)
```

File: lib/msal-react/docs/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+1. [Determinting whether a user is authenticated](#determining-whether-a-user-is-authenticated)
+1. [Signing a user in](#signing-a-user-in)
+1. [Acquiring an access token](#acquring-an-access-token)
```
Comment (jo-arroyo) [inline lib/msal-react/docs/getting-started.md:6]:
```suggestion
1. [Acquiring an access token](#acquiring-an-access-token)
```

File: lib/msal-react/docs/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+const AppProvider = () => (
+    <MsalProvider instance={pca}>
+        <App />
+    </MsalProvider>
+);
+
+ReactDOM.render(<AppProvider />, document.getElementById("root"));
+```
+
+All components underneath `MsalProvider` will have access to the `PublicClientApplication` instance via context as well as all hooks and components provided by `msal-react`
```
Comment (jo-arroyo) [inline lib/msal-react/docs/getting-started.md:40]:
```suggestion
All components underneath `MsalProvider` will have access to the `PublicClientApplication` instance via context as well as all hooks and components provided by `msal-react`.
```

File: lib/msal-react/docs/getting-started.md [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+                <button onClick={() => instance.loginPopup()}>Login</button>
+            </>
+        );
+    }
+}
+```
+
+#### Consuming the raw context
+
+If you are using a class component and can't use hooks you can consume the raw msal context through `MsalContext`
```
Comment (jo-arroyo) [inline lib/msal-react/docs/getting-started.md:158]:
```suggestion
If you are using a class component and can't use hooks you can consume the raw msal context through `MsalContext`.
```

File: samples/msal-react-samples/default/README.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
+This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
```
Comment (jo-arroyo) [inline samples/msal-react-samples/default/README.md:1]: Unsure if the React sample needs a specific readme on how to use it?

**Review by jo-arroyo — COMMENTED** [review]
Just a few nits! Otherwise LGTM

File: samples/msal-react-samples/default/README.md (hunk 6 above)
Comment (tnorling) [reply samples/msal-react-samples/default/README.md:1]: Yes! Thanks for calling this out, forgot about this

Comment (tnorling) [reply samples/msal-react-samples/default/README.md]: Done!

File: lib/msal-react/README.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+The `@azure/msal-react` package described by the code in this folder uses the [`@azure/msal-browser` package](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/lib/msal-browser) as a peer dependency to enable authentication in Javascript Single-Page Applications without backend servers. This version of the library uses the OAuth 2.0 Authorization Code Flow with PKCE. To read more about this protocol, as well as the differences between implicit flow and authorization code flow, see the section [below](#implicit-flow-vs-authorization-code-flow-with-pkce).
```
Comment (tnorling) [inline lib/msal-react/README.md:23]:
```suggestion
The `@azure/msal-react` package described by the code in this folder uses the [`@azure/msal-browser` package](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/lib/msal-browser) as a peer dependency to enable authentication in Javascript Single-Page Applications without backend servers. This version of the library uses the OAuth 2.0 Authorization Code Flow with PKCE. To read more about this protocol, as well as the differences between implicit flow and authorization code flow, see the section in the [msal-browser readme](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/README.md#implicit-flow-vs-authorization-code-flow-with-pkce).
```

**Review by jasonnutter — APPROVED** [review]

File: lib/msal-react/src/MsalProvider.tsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    EventMessage, InteractionType
+} from "@azure/msal-browser";
+import { MsalContext, IMsalContext } from "./MsalContext";
+import { InteractionStatus } from "./utils/Constants";
+
+export type MsalProviderProps = PropsWithChildren<{
+    instance: IPublicClientApplication;
+}>;
+
+export function MsalProvider({instance, children}: MsalProviderProps) {
```
Comment (pkanher617) [inline lib/msal-react/src/MsalProvider.tsx:20]: Add a return type here.

**Review by pkanher617 — APPROVED** [review]
Would like to fix some of the linter/ts errors now rather than later, but otherwise looks good. Excited to see this released!

File: lib/msal-react/docs/getting-started.md [hunk 9]
```diff
@@ -0,0 +1,222 @@
+# Getting Started
+
+1. [Initialization](#Initialization)
```
Comment (jo-arroyo) [inline lib/msal-react/docs/getting-started.md:3]:
```suggestion
1. [Initialization](#initialization)
```

Comment (jo-arroyo) [reply lib/msal-react/docs/getting-started.md:3]: I think it doesn't link properly if capitalized?

**Review by jo-arroyo — APPROVED** [review]
LGTM!

File: lib/msal-react/src/MsalProvider.tsx (hunk 8 above)
Comment (tnorling) [reply lib/msal-react/src/MsalProvider.tsx]: Done

File: change/@azure-msal-browser-2020-11-06-11-43-57-msal-react-feature-branch.json [hunk 10]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "patch",
+  "comment": "Export stubbed PCA instance (#2540)",
+  "packageName": "@azure/msal-browser",
+  "email": "thomas.norling@microsoft.com",
+  "dependentChangeType": "patch",
```
Comment (tnorling) [inline change/@azure-msal-browser-2020-11-06-11-43-57-msal-react-feature-branch.json:6]:
```suggestion
  "dependentChangeType": "none",
```

File: change/@azure-msal-react-2020-11-06-11-43-57-msal-react-feature-branch.json [hunk 11]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "prerelease",
```
Comment (tnorling) [inline change/@azure-msal-react-2020-11-06-11-43-57-msal-react-feature-branch.json:2]:
```suggestion
  "type": "none",
```

File: change/@azure-msal-react-2020-11-06-11-43-57-msal-react-feature-branch.json [hunk 12]
```diff
@@ -0,0 +1,8 @@
...
+  "comment": "msal-react first alpha release (#2540)",
+  "packageName": "@azure/msal-react",
+  "email": "thomas.norling@microsoft.com",
+  "dependentChangeType": "patch",
```
Comment (tnorling) [inline change/@azure-msal-react-2020-11-06-11-43-57-msal-react-feature-branch.json:6]:
```suggestion
  "dependentChangeType": "none",
```

**Pull request merged**